text/3453-f16-and-f128.md

```diff
@@ -0,0 +1,128 @@
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## `f16` type
+
+`f16` consists of 1 bit of sign, 5 bits of exponent, 10 bits of mantissa. It is exactly equivalent to the 16-bit IEEE 754 binary16 [Half-precision floating-point format](https://en.wikipedia.org/wiki/Half-precision_floating-point_format).
+
+The following traits will be implemented for conversion between `f16` and other types:
+
+```rust
+impl From<f16> for f32 { /* ... */ }
+impl From<f16> for f64 { /* ... */ }
+impl From<bool> for f16 { /* ... */ }
+impl From<u8> for f16 { /* ... */ }
+impl From<i8> for f16 { /* ... */ }
+```
+
+`f16` will generate `half` type in LLVM IR. It is also equivalent to C++ `std::float16_t`, C `_Float16`, and GCC `__fp16`.
+
+On the hardware level, `f16` can be accelerated on RISC-V via [the Zfh or Zfhmin extensions](https://five-embeddev.com/riscv-isa-manual/latest/zfh.html), on x86 with AVX-512 via [the FP16 instruction set](https://en.wikipedia.org/wiki/AVX-512#FP16), on [some Arm platforms](https://developer.arm.com/documentation/100067/0607/Other-Compiler-specific-Features/Half-precision-floating-point-number-format), and on PowerISA via [VSX on PowerISA v3.1B and later](https://files.openpower.foundation/s/dAYSdGzTfW4j2r2). Most platforms do not have hardware support and therefore will need to use a software implementation.
+
+## `f128` type
+
+`f128` consists of 1 bit of sign, 15 bits of exponent, 112 bits of mantissa. It is exactly equivalent to the 128-bit IEEE 754 binary128 [Quadruple-precision floating-point format](https://en.wikipedia.org/wiki/Quadruple-precision_floating-point_format).
+
+The following traits will be implemented for conversion between `f128` and other types:
+
+```rust
+impl From<f16> for f128 { /* ... */ }
+impl From<f32> for f128 { /* ... */ }
+impl From<f64> for f128 { /* ... */ }
+impl From<bool> for f128 { /* ... */ }
+impl From<u8> for f128 { /* ... */ }
+impl From<i8> for f128 { /* ... */ }
+impl From<u16> for f128 { /* ... */ }
+impl From<i16> for f128 { /* ... */ }
+impl From<u32> for f128 { /* ... */ }
+impl From<i32> for f128 { /* ... */ }
+impl From<u64> for f128 { /* ... */ }
+impl From<i64> for f128 { /* ... */ }
+```
+
+`f128` will generate `fp128` type in LLVM IR. It is also equivalent to C++ `std::float128_t`, C `_Float128`, and GCC `__float128`.
+
+On the hardware level, `f128` can be accelerated on RISC-V via [the Q extension](https://five-embeddev.com/riscv-isa-manual/latest/q.html), and on PowerISA via [BFP128, an optional part of PowerISA v3.0C and later](https://files.openpower.foundation/s/XXFoRATEzSFtdG8). Most platforms do not have hardware support and therefore will need to use a software implementation.
```

Comment:
It feels worth mentioning a bit on the fact that `as` casts are available for `f16` and `f128` similarly to how they are for existing floats. This will allow for truncating conversions, particularly from `f32` and `f64` to `f16`.